test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -118,5 +135,111 @@ public void ChangePointDetectionWithSeasonality()
                 Assert.Equal(expectedValues[index++], row.Change[3], precision: 7);  // Martingale score
             }
         }
+
+        [Fact]
+        public void ChangePointDetectionWithSeasonalityPredictionEngineNoColumn()
+        {
+            const int ChangeHistorySize = 10;
+            const int SeasonalitySize = 10;
+            const int NumberOfSeasonsInTraining = 5;
+            const int MaxTrainingSize = NumberOfSeasonsInTraining * SeasonalitySize;
+
+            List<Data> data = new List<Data>();
+
+            var ml = new MLContext(seed: 1, conc: 1);
+            var dataView = ml.CreateStreamingDataView(data);
+
+            for (int j = 0; j < NumberOfSeasonsInTraining; j++)
+                for (int i = 0; i < SeasonalitySize; i++)
+                    data.Add(new Data(i));
+
+            for (int i = 0; i < ChangeHistorySize; i++)
+                data.Add(new Data(i * 100));
+
+
+            // Pipeline.
+            var pipeline = ml.Transforms.Text.FeaturizeText("Text", "Text_Featurized")
+                .Append(new SsaChangePointEstimator(ml, new SsaChangePointDetector.Arguments()
+                {
+                    Confidence = 95,
+                    Source = "Value",
+                    Name = "Change",
+                    ChangeHistoryLength = ChangeHistorySize,
+                    TrainingWindowSize = MaxTrainingSize,
+                    SeasonalWindowSize = SeasonalitySize
+                }));
+
+            // Train.
+            var model = pipeline.Fit(dataView);
+
+            //Predict.
+            var engine = model.CreateTimeSeriesPredictionFunction<Data, Prediction1>(ml);
+            //Even though time series column is not requested it will pass the observation through time series transform.
+            var prediction = engine.Predict(new Data(1));
+            Assert.Equal(-1, prediction.Random);
+            prediction = engine.Predict(new Data(2));
+            Assert.Equal(-1, prediction.Random);
```

Comment:
I didnt get how you check that observations are passing through timeseries transform

Reply:
Debugger but we can update this test to check-point before and after, I have made this change in the below test.

---
In reply to: [236521589](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236521589) [](ancestors = 236521589)

Reply:
Debugger? its supposed to be a unit test :) with no debugger attached, would it be possible to this check in code?

---
In reply to: [236804680](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236804680) [](ancestors = 236804680,236521589)

Reply:
Done,

---
In reply to: [236831759](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236831759) [](ancestors = 236831759,236804680,236521589)